servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AbstractStreamingHttpConnection.java

```diff
@@ -119,8 +119,20 @@ public final HttpConnectionContext connectionContext() {
                                 .scanWith(HeaderUtils::insertTrailersMapper);
                 addRequestTransferEncodingIfNecessary(request);
             }
-            return strategy.invokeClient(executionContext.executor(), flatRequest,
-                    determineFlushStrategyForApi(request), this).subscribeShareContext();
+
+            if (strategy.isSendOffloaded()) {
+                flatRequest = flatRequest.subscribeOn(executionContext.executor());
+            }
+            Single<StreamingHttpResponse> resp = invokeClient(flatRequest, determineFlushStrategyForApi(request));
+            if (strategy.isMetadataReceiveOffloaded()) {
+                resp = resp.publishOn(executionContext.executor());
```

Comment:
I would expect each of the `subscribeOn` and `publishOn` in this method to be using something like `Thread.currentThread() instanceof IoThreadFactory.IoThread` for the `BooleanSupplier`, wdyt?

Reply:
Seems reasonable.